frontend/src/pages/login/registerPage.tsx

```diff
@@ -0,0 +1,210 @@
+import React, { useState } from 'react'
+import {
+  IonBackButton,
+  IonButton,
+  IonButtons,
+  IonCard,
+  IonCardContent,
+  IonCheckbox,
+  IonContent,
+  IonHeader,
+  IonInput,
+  IonItem,
+  IonItemGroup,
+  IonLabel,
+  IonModal,
+  IonPage,
+  IonSelect,
+  IonSelectOption,
+  IonSpinner,
+  IonTitle,
+  IonToolbar,
+  useIonToast,
+} from '@ionic/react'
+import AuthState from '../../model/authState'
+import institutions from '../../data/institutions'
+import './login.css'
+import { useRegister } from '../../model/registerContext'
+import Auth from '@aws-amplify/auth'
+
+interface RegisterPageProps {
+  setAuthState(authState: AuthState): void
+}
+
+const RegisterPage: React.FC<RegisterPageProps> = ({ setAuthState }) => {
+  const [forename, setForename] = useState('')
+  const [surname, setSurname] = useState('')
+  const [institution, setInstitution] = useState('')
+  const [agreedToTerms, setAgreedToTerms] = useState(false)
```

Comment:
Why not working with one useState?
```suggestion
  const [userProps, setUserProps] = useState({forename, surname, institution, agreedToTerms})
```

```javascript
setUserProps(prevState => {...prevState, forename:"myName is"})
```

Reply:
Because I didn't think of it! Changed!